src/OrchardCore/OrchardCore.Setup.Core/SetupService.cs

```diff
@@ -202,32 +199,21 @@ private async Task<string> SetupInternalAsync(SetupContext context)
             {
                 await shellContext.CreateScope().UsingServiceScopeAsync(async scope =>
                 {
-                    IStore store;
-
                     try
                     {
-                        store = scope.ServiceProvider.GetRequiredService<IStore>();
+                        // Create the "minimum shell descriptor"
+                        await scope
+                            .ServiceProvider
+                            .GetService<IShellDescriptorManager>()
+                            .UpdateShellDescriptorAsync(0,
+                                shellContext.Blueprint.Descriptor.Features);
                     }
                     catch (Exception e)
                     {
-                        // Tables already exist or database was not found
-
-                        // The issue is that the user creation needs the tables to be present,
-                        // if the user information is not valid, the next POST will try to recreate the
-                        // tables. The tables should be rolled back if one of the steps is invalid,
-                        // unless the recipe is executing?
-
-                        _logger.LogError(e, "An error occurred while initializing the datastore.");
-                        context.Errors.Add(String.Empty, S["An error occurred while initializing the datastore: {0}", e.Message]);
+                        _logger.LogError(e, "An error occurred while initializing the shell descriptor manager.");
+                        context.Errors.Add(String.Empty, S["An error occurred while initializing the shell descriptor manager: {0}", e.Message]);
```

Comment:
Maybe here the error message is too technical.

I suggest to keep the previous message (already translated ;)) because we still do a first data store initialization before a first update of the shell descriptor in this data store that checks if such a table (with a given serialNumber) already exists, and anyway we pass `e.Message`, but as you want.

Otherwise, maybe something like `Failed to update the tenant descriptor to the data store.`